core/proto/core.proto

```diff
@@ -86,7 +86,6 @@ message Registration {
   repeated string contact = 3;
   bool contactsPresent = 4;
   string agreement = 5;
-  bytes initialIP = 6;
```

Comment:
> Should field 6 be reserved?

It actually is, if you look just under the message definition you'll see:

```proto
message Registration {
  // Next unused field number: 10
```

This is how we ensure that we never reuse a proto field in future releases.

> Is this deployable in a single release as-is? It seems like if a WFE gets updated before an RA, the RA might reject new registrations due to incomplete gRPC fields.

Scenario 1, an updated sender (without `InitialIP`) communicates with outdated receiver (expects `InitialIP`).

Outdated receiver treats `InitialIP` as missing and defaults it to empty bytes. This is acceptable provided we've ignored the zero-values, which this PR accomplishes by deleting zero-value checks on the first lines of `NewRegistration()` and `UpdateRegistration()`.

Scenario 2, an outdated sender (with `InitialIP`) communicates with an updated receiver (does not expect `InitialIP`):

Updated receiver safely ignores the unknown InitialIP field. Protobuf is designed specifically to ignore unknown fields because it makes the process of rolling out new fields simpler.

Reply:
I think James was asking if it should be explicitly marked as reserved like we do on the line directly below this. It's not strictly necessary to do so, but it's not quite accurate to say that it is reserved by virtue of the comment -- the protoc compiler's behavior is different depending on whether an actual `reserved 6;` line exists. And regardless, it's a good idea to match the style of the surrounding code, which does take advantage of the "reserved" keyword.

I agree that Scenario 2 is safe. In Scenario 1, the non-updated receiver will not have had its zero-value checks removed yet, so it will error out with "incomplete gRPC request". I think the right fix here is simply to land the SA change first, so it no longer matters whether the WFE and RA are passing around IP addresses or not, then land the WFE and RA changes. Good catch, James, I totally missed this.

Reply:
I agree with the above, but there are nil checks on the server side for _both_ the SA and the RA. I think the "land first" code would be "remove the nil check on initialIP in both places."

Alternately, to land most of this code in one change, we could continue passing initialIP but make WFE responsible for defaulting it to 0.0.0.0 instead of SA. I believe that would be deployable as a single change.

Reply:
> Outdated receiver treats InitialIP as missing and defaults it to empty bytes. This is acceptable provided we've ignored the zero-values...

Lol, yeah, I can't believe didn't spot this. Good catch @jprenken.

Reply:
> Alternately, to land most of this code in one change, we could continue passing initialIP but make WFE responsible for defaulting it to 0.0.0.0 instead of SA. I believe that would be deployable as a single change.

This is a great idea, thanks @jsha.